go-controller/pkg/ovn/secondary_layer2_network_controller.go

```diff
@@ -594,8 +594,14 @@ func (oc *SecondaryLayer2NetworkController) addUpdateLocalNodeEvent(node *corev1
 				} else {
 					if !util.IsPodNetworkAdvertisedAtNode(oc, node.Name) {
 						err = oc.addUDNClusterSubnetEgressSNAT(gwConfig.hostSubnets, gwManager.gwRouterName)
+						if err == nil && util.IsRouteAdvertisementsEnabled() {
+							err = oc.deleteAdvertisedUDNIsolation(node.Name)
+						}
 					} else {
 						err = oc.deleteUDNClusterSubnetEgressSNAT(gwConfig.hostSubnets, gwManager.gwRouterName)
+						if err == nil {
```

Comment:
no need for feature enablement flag?

Reply:
no need, it is an else block for ` !util.IsPodNetworkAdvertisedAtNode(oc, node.Name)`